src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,204 @@
+        for (Product product : order.getProducts()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, order.getId());
+            preparedStatement.setLong(2, product.getId());
+            result += preparedStatement.executeUpdate();
+        }
+        return result;
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders"
+                        + " JOIN orders_products ON orders.order_id"
+                        + " = orders_products.orders_id"
+                        + " JOIN products ON products.product_id"
+                        + " = orders_products.products_id"
+                        + " where orders.order_id = ? ";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet sqlResult = preparedStatement.executeQuery();
+            if (sqlResult.next()) {
+                Order order = parseResult(sqlResult);
+                return Optional.of(order);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get order by id = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    private Order parseResult(ResultSet sqlResult) throws SQLException {
+        Long id = sqlResult.getLong("order_id");
+        Long userId = sqlResult.getLong("user_id");
+        List<Product> productList = new ArrayList<>();
+        do {
+            productList.add(parseProduct(sqlResult));
+        } while (sqlResult.next());
+        return new Order(id, userId, productList);
+    }
+
+    private Product parseProduct(ResultSet sqlResult) throws SQLException {
+        Long productId = sqlResult.getLong("product_id");
+        String name = sqlResult.getString("name");
+        Double price = sqlResult.getDouble("price");
+        return new Product(productId, name, price);
+    }
+
+    @Override
+    public Order update(Order order) {
+        String query = "DELETE FROM orders_products"
+                        + " WHERE orders_id = ?";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, order.getId());
+            preparedStatement.executeUpdate();
+            insertProducts(order, connection);
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't update order for user = " + order.getUserId(), e);
+        }
+        return order;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE orders"
+                        + " SET deleted = true"
+                        + " WHERE order_id = ? and deleted = false";
```

Comment:
why do you have "...and deleted = false"? It's not necessary check

Reply:
Ok. Done